### PR TITLE
Fix bugs #991 in HRZones

### PR DESCRIPTION
**Pull request by sv2ramanan**
These changes address bug/issue #991 . 

Part 1 addresses "The UI should not allow changing the upper limit."

1. HRZonesActivity.java: 
  Make the high range of HR Zones uneditable.
   Map the low end of a zone to the high end of the previous zone on-the-fly
   A first attempt to make sure the zones are ordered (increasing)

Part 2  fixes a bug in zones: "saved HR zone is +2 to the selected. If I choose zone 1, after closing and reopening it becomes zone 3. And zones 4 and 5 become empty.". This behavior seems to derive from different zero- versus one-based numbering: 1 for the zones (for users) and 0 programmatically. This results in changes of 2 in HR zones when editing zones as targets of intervals.

2. StepButton.java:
   When HR is the target in an interval, the zone changes by +2 each time it is re-edited.
   This is fixed by subtracting 2 from the results of the match of hrZones from its adapter
   This is probably not the best way to write the solution, but it prevents app crashes as well.

### CODE REVIEW

File: app/src/main/org/runnerup/view/StepButton.java [hunk 1]
```diff
@@ -300,8 +300,11 @@ public int preSetValue(int newValue) throws IllegalArgumentException {
                         targetHrz.setEnabled(true);
                         targetHrz.setVisibility(View.VISIBLE);
                         if (target != null) {
-                            targetHrz.setValue(hrZonesAdapter.hrZones.match(target.minValue,
-                                    target.maxValue));
+                            int matchedZone=hrZonesAdapter.hrZones.match(target.minValue,
+                                    target.maxValue)-2;
+                            if (matchedZone<0)
```
Comment (gerhardol) [inline app/src/main/org/runnerup/view/StepButton.java:305]: The code base is inconsistent, but I want braces at all if and spaces around operators

**Review by gerhardol — COMMENTED** [review]
Seem OK, have briefly played around with it

How to handle upper for zone 5?
It is probably OK to lock it to MHR
It is possible to set lower for zone 5 to > than MHR though... (just increase MHR?)

**Comment by sv2ramanan** [comment]
1. Fixed braces and spaces in code
2. Set low HR for zone 5 to be maxHR-1 at most
3. Set low HR for any zone to be maxHR - 1 - (noZones-zone), so each Zone has 1 HR beat range at least

File: app/src/main/org/runnerup/view/StepButton.java [hunk 2]
```diff
@@ -300,16 +300,19 @@ public int preSetValue(int newValue) throws IllegalArgumentException {
                         targetHrz.setEnabled(true);
                         targetHrz.setVisibility(View.VISIBLE);
                         if (target != null) {
-                            targetHrz.setValue(hrZonesAdapter.hrZones.match(target.minValue,
-                                    target.maxValue));
+                            int matchedZone = hrZonesAdapter.hrZones.match(target.minValue,
```
Comment (gerhardol) [inline app/src/main/org/runnerup/view/StepButton.java:303]: Please add a comment here

**Review by gerhardol — APPROVED** [review]
It is still possible to set zones in wrong order, but it is better than before
Do you want to try fix it or merge as is?

After adding the comment, do you want me to squash merge or clean up the commits yourself first?

**Pull request closed without merging**